lib/schema.js

```diff
@@ -2638,6 +2638,9 @@ Schema.prototype._getSchema = function(path) {
       // Re: gh-5628, because `schema.path()` doesn't take $ into account.
       parts[i] = '0';
     }
+    if (/^\d+$/.test(parts[i])) {
```

Comment:
extracting a regex into a variable outside the calling function should be more performant.